src/http-adapter.js

```diff
@@ -1,148 +0,0 @@
-/***
- * HttpAdapter
- * @module
- */
-import {ajax} from 'jquery'
-
-/**
- * Adapter which works with data over HTTP, based on the options that
- * are passed to it's constructor. This class is responsible for
- * any CRUD operations that need to be carried out with your {@link https://jsonapi.org/ JSON:API} service
- * from within the Backbone.Store.
- * @param {Object} options -  An object that contains a property, `urlPrefix`,
- * that defines the REST service that will be returning a {@link https://jsonapi.org/ JSON:API} response
- */
-class HttpAdapter {
-  constructor (options = {}) {
-    this.urlPrefix = options.urlPrefix
-
-    this.serializeRequests = false
-    this._outstandingRequests = new Set()
-  }
-
-  buildUrl (type, id) {
-    let idPath = `/${id}`
-    let typePath = `/${type}`
-    let path = this.urlPrefix || ''
-
-    path += typePath
-
-    if (id != null) {
-      path += idPath
-    }
-
-    return path + '/'
-  }
-
-  /**
-   * Get entity by link.
-   * @private
-   * @param {string} link - Link to entity.
-   * @returns {Promise} Promise for fetched data.
-   */
-  get (link, query) {
-    return this._ajax('GET', link, query)
-      .then(body => JSON.parse(body))
-  }
-
-  /**
-   * Create entity.
-   * @private
-   * @param {string} link - Entity url.
-   * @param {object} attributes - Data to create entity with.
-   * @returns {Promise} Promise for created data.
-   */
-  create (link, payload) {
-    return this._ajax('POST', link, payload)
-      .then(body => body && JSON.parse(body))
-  }
-
-  /**
-   * Update entity.
-   * @private
-   * @param {string} link - Entity url.
-   * @param {object} attributes - Data to update entity with.
-   * @returns {Promise} Promise for updated data.
-   */
-  update (link, payload) {
-    return this._ajax('PATCH', link, payload)
-      .then(body => JSON.parse(body))
-  }
-
-  /**
-   * Destroy entity.
-   * @private
-   * @param {string} link - Entity self link.
-   * @returns {Promise} Promise for destroy.
-   */
-  destroy (link) {
-    return this._ajax('DELETE', link)
-  }
-
-  _ajax (type, url, data) {
-    let headers = {
-      'Accept': 'application/vnd.api+json',
-      'Content-Type': 'application/vnd.api+json'
-    }
-
-    // Stringify data before any async stuff, just in case it's accidentally a mutable object (e.g.
-    // some instrumented Vue data)
-    if (data && ['PATCH', 'POST'].indexOf(type) > -1) {
-      data = JSON.stringify(data)
-    }
-
-    let promise
```

Comment:
94-112 moved to [`_checkSerializeRequests`](https://github.com/groveco/backbone.store/pull/76/files#diff-9d8a63065d5db23b88af37023f10a547R134) in the base `HttpAdapter`